### ISSUE TITLE
create a index but return error

### ISSUE THREAD

**Issue by xushuhui** [meilisearch/meilisearch#3965]
**Describe the bug**
Thanks your work in meilisearch
I find a problem when I use go sdk call meilisearch，it create a index successful but return a error.

**To Reproduce**

```go
func TestMilisearch(t *testing.T) {
	client := meilisearch.NewClient(meilisearch.ClientConfig{
		Host: "xx:7700",
		APIKey: "",
	  })
	res,err :=   client.CreateIndex(&meilisearch.IndexConfig{
		Uid: "news_v1",
	  })
	 t.Log(res,err)
}
```
5. See error
```
<nil> unaccepted status code found: 201 expected: [202], MeilisearchApiError Message:  (path "POST /indexes" with method "CreateIndex")
```
But I see created index `news_v1` in dashboard，It is error or normal?
**Expected behavior**
no error 

**Screenshots**
<img width="1324" alt="image" src="https://github.com/meilisearch/meilisearch/assets/13704743/424aaa3d-e3ae-4d72-a182-b7cca6bd1e34">

**Meilisearch version:**
0.24.0

**Additional context**
Additional information that may be relevant to the issue.
[e.g. architecture, device, OS, browser]
go version 1.20
github.com/meilisearch/meilisearch-go v0.25.0

**Comment by curquiza** [comment]
Hello @xushuhui 

I see you use v0.24.0. You have to know Meilisearch v0.24.0 is a really old version of Meilisearch (more than a year and a half), and is not even a stable version of Meilisearch. We released v1.0.0 in February 2023. The latest one we have just released is v1.3.0 of Meilisearch.

You can update to the latest version of Meilisearch by following this guide: https://www.meilisearch.com/docs/learn/update_and_migration/updating

To ease your future update of Meilisearch, your deployment and settings management, you can also use our [cloud offer](https://www.meilisearch.com/cloud?utm_campaign=oss&utm_source=github&utm_medium=meilisearch) to manage Meilisearch in an easier way. The migration to another version of Meilisearch can only be done with one click.

**Comment by xushuhui** [comment]
@curquiza  I update to the latest version v1.3.0，it solve my problem，Thank you
